fabtests/functional/unmap_mem.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2013-2019 Intel Corporation.  All rights reserved.
+ *
+ * This software is available to you under the BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <sys/mman.h>
+
+#include <shared.h>
+
+
+static int map_tx_buf(void)
+{
+	/* Try to get the mapping at the same virtual address */
+	tx_buf = mmap(tx_buf, opts.transfer_size, PROT_READ | PROT_WRITE,
+		      MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+	if (!tx_buf) {
+		FT_PRINTERR("mmap", -errno);
+		return -errno;
+	}
+	return 0;
+}
+
+static void unmap_tx_buf(void)
+{
+	int ret;
+
+	ret = madvise(tx_buf, opts.transfer_size, MADV_DONTNEED);
+	if (ret)
+		FT_PRINTERR("madvise", -errno);
+
+	ret = munmap(tx_buf, opts.transfer_size);
+	if (ret)
+		FT_PRINTERR("munmap", -errno);
+}
+
+static int pingpong(void)
```

Comment:
Does this test assume running on loopback?

Reply:
No - it can run between systems.  However, it requires out-of-band synchronization because it usurps the tx buffer.